eip/EIPS/eip-4337.md

```diff
@@ -184,6 +184,8 @@ We extend the entry point logic to support **paymasters** that can sponsor trans
 
 ![](../assets/eip-4337/image2.png)
 
+First, a paymaster must issue `addStake()` to lock some eth for a period of time. This period is defined by the EntryPoint. Note that the amount staked is not defined on-chain, and MUST be
+validated by nodes during the UserOperation validation phase. The staked value MUST be above 1000 USD in native tokens (and SHOULD be larger, to compensate for native token price fluctuations)
```

Comment:
I'm not sure if we can really define it that way.  Fluctuations can be a major issue for autonomous (unmanaged) paymasters.  E.g. a TokenPaymaster that remains balanced by swapping in Uniswap, and has no owner.  Someone deposited the stake once, and is no longer around.  Suddenly the native currency price plunges, and the paymaster becomes unreachable.

Therefore I don't think the stake can be denominated in USD and tracked automatically.  It has to be a value that makes sense for the current chain, but should remain constant unless there's an extreme price drop that warrants a manual change.  E.g. make it 1 eth, but adjust it manually if eth drops to $500 or reaches $5000.

We could mitigate the issue by keeping a long term reputation for paymasters rather than just the sliding scale.  If a paymaster has been active for a year with no issues, and then its stake becomes insufficient due to eth price dropping, it makes sense to let it keep running.  However, this would complicate things.  Probably safer to just have a default value denominated in the chain's native currency, and adjust it manually in extreme situations.  

At the very least, I wouldn't use the word "MUST", but advise that the network should use a value equivalent to that.

WDYT?